src/classes/frModel.cls

```diff
@@ -0,0 +1,71 @@
+public abstract class frModel {
+	protected abstract SObject getObject();
+    public abstract List<frMapping__c> getMappings();
+
+    public virtual void populateFromRequest(Map<String, Object> request) {
+    	Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(getObject().getSObjectType().getDescribe().getName());
+
+        Map<String, frMapping__c> frNameToSfName = new Map<String, frMapping__c>();
+        for(frMapping__c mapping : getMappings()) {
+            frNameToSfName.put(mapping.fr_Name__c, mapping);
```

Comment:
Are you opposed to logging some message here if the mapping.fr_Name__c is empty? Something went wrong when I was setting up my package and all of my mappings had no funraise fields, but it worked on the setup page and SFDC did not complain or generate any logs when I sent over a donation. It also didn't actually create the donation though so there is at least some indication that things went wrong